lib/chef/knife/core/status_presenter.rb

```diff
@@ -100,7 +100,14 @@ def summarize(list)
             fqdn = (node[:ec2] && node[:ec2][:public_hostname]) || node[:fqdn]
             name = node["name"] || node.name
 
-            run_list = (node["run_list"]).to_s if config[:run_list]
+            if config[:run_list]
+              run_list = node["run_list"]
```

Comment:
why not make this the else to the if below. If long_output is set you're setting the value and then setting the value again.